agent/config/builder.go

```diff
@@ -1885,13 +1898,35 @@ func uintVal(v *uint) uint {
 	return *v
 }
 
+func uint32Val(v *uint32) uint32 {
+	if v == nil {
+		return 0
+	}
+	return *v
+}
+
 func uint64Val(v *uint64) uint64 {
 	if v == nil {
 		return 0
 	}
 	return *v
 }
 
+// Expect an octal permissions string, e.g. 0644
+func (b *builder) unixPermissionsVal(name string, v *string) uint32 {
+
+	if v == nil {
+		return 0
+	}
+	if strings.HasPrefix(*v, "0") {
+		if mode, err := strconv.ParseUint(*v, 0, 32); err == nil {
+			return uint32(mode)
+		}
+	}
```

Comment:
Not all octal numbers start with `0`. You can do `chmod 1755` and set the higher positional values if you want. You could just safely assume this is 100% of the time an octal number and switch it up to just:

```suggestion
	if mode, err := strconv.ParseUint(*v, 8, 32); err == nil {
		return uint32(mode)
	}
```